Package: vnc4
Version: 4.1.1+X4.3.0-6
Severity: wishlist
Tags: patch

When building 'vnc4' on ppc64/unstable, I get the following error:

gcc -O2 -fsigned-char    -fno-merge-constants -I../../../programs/Xserver/fb -I../../../programs/Xserver/mi -I../../../programs/Xserver/include 	      -I../../../exports/include/X11 	      -I../../../include/fonts -I../../../programs/Xserver/hw/xfree86/common  	      -I../../../programs/Xserver/render -I../../../include/extensions -I../../../programs/Xserver/Xext  -I../../.. -I../../../exports/include -I/usr/X11R6/include  -Dlinux -D__powerpc__ -D_POSIX_C_SOURCE=199309L 				-D_POSIX_SOURCE -D_XOPEN_SOURCE 				-D_BSD_SOURCE -D_SVID_SOURCE 				 -D_GNU_SOURCE 				 -DSHAPE -DXINPUT -DXKB -DLBX -DXAPPGROUP 	-DXCSECURITY -DTOGCUP  	-DXF86BIGFONT -DDPMSExtension 	  -DPANORAMIX 	 -DRENDER -DRANDR -DGCCUSESGAS -DAVOID_GLYPHBLT -DPIXPRIV -DSINGLEDEPTH 				-DXFreeXDGA -DXvExtension 				-DXFree86LOADER  -DXFree86Server 				-DXF86VIDMODE 				-DXvMCExtension 				 				-DSMART_SCHEDULE 				-DBUILDDEBUG -DXResExtension 				-DX_BYTE_ORDER=X_BIG_ENDIAN -DNDEBUG  -DFUNCPROTO=15 -DNARROWPROTO  -DIN_MODULE -DXFree86Module    -c fbblt.c
fbblt.c: In function 'fbBlt':
fbblt.c:136: error: duplicate case value

With the attached patch 'vnc4' can be compiled on ppc64.

Regards
Andreas Jochens

diff -urN ../tmp-orig/vnc4-4.1.1+X4.3.0/unix/xc/config/cf/linux.cf ./unix/xc/config/cf/linux.cf
--- ../tmp-orig/vnc4-4.1.1+X4.3.0/unix/xc/config/cf/linux.cf	2006-04-21 11:53:04.000000000 +0000
+++ ./unix/xc/config/cf/linux.cf	2006-04-21 09:02:32.000000000 +0000
@@ -818,7 +818,11 @@
 # endif
 # define LinuxMachineDefines	-D__powerpc__
 # define ServerOSDefines	XFree86ServerOSDefines -DDDXTIME -DPART_NET
+# ifdef __powerpc64__
+# define ServerExtraDefines	-DGCCUSESGAS XFree86ServerDefines -D_XSERVER64
+# else
 # define ServerExtraDefines	-DGCCUSESGAS XFree86ServerDefines
+# endif
 #endif /* PpcArchitecture */
 
 #ifdef s390Architecture
diff -urN ../tmp-orig/vnc4-4.1.1+X4.3.0/unix/xc/include/Xmd.h ./unix/xc/include/Xmd.h
--- ../tmp-orig/vnc4-4.1.1+X4.3.0/unix/xc/include/Xmd.h	2002-05-31 18:45:39.000000000 +0000
+++ ./unix/xc/include/Xmd.h	2006-04-21 10:18:20.000000000 +0000
@@ -60,7 +60,7 @@
 #endif
 #if defined(__alpha) || defined(__alpha__) || \
     defined(__ia64__) || defined(ia64) || \
-    defined(__sparc64__) || \
+    defined(__sparc64__) || defined(__powerpc64__) || \
     defined(__s390x__) || \
     (defined(__hppa__) && defined(__LP64__)) || \
     defined(__x86_64__) || defined(x86_64)


